plugin-core/src/main/java/appland/actions/PluginStatus.java

```diff
@@ -0,0 +1,114 @@
+package appland.actions;
+
+import appland.cli.AppLandDownloadService;
+import appland.cli.CliTool;
+import appland.javaAgent.JavaAgentStatus;
+import com.intellij.openapi.actionSystem.AnAction;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.actionSystem.UpdateInBackground;
+import com.intellij.openapi.application.WriteAction;
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.openapi.fileEditor.FileEditorManager;
+import com.intellij.openapi.progress.PerformInBackgroundOption;
+import com.intellij.openapi.progress.ProgressIndicator;
+import com.intellij.openapi.progress.Task;
+import com.intellij.openapi.project.DumbAware;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.project.ProjectUtil;
+import com.intellij.openapi.vfs.VfsUtil;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.util.concurrency.annotations.RequiresBackgroundThread;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.io.IOException;
+import java.util.Objects;
+
+
+public class PluginStatus extends AnAction implements UpdateInBackground, DumbAware {
+    private static final Logger LOG = Logger.getInstance(PluginStatus.class);
+    private static final String STATUS_REPORT_FILENAME = "appmap_status_report.md";
+
+    @Override
+    public void actionPerformed(@NotNull AnActionEvent event) {
+        createStatusReport(Objects.requireNonNull(event.getProject()));
```

Comment:
If there's no requirement to keep it static, I'd just inline `createStatusReport` here.

Reply:
Done! :+1: